### PR TITLE
feat(OnyxTable): add row and column hover styles

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #838

Add table row and column hover styles (plain CSS implementation).

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 1]
```diff
@@ -124,10 +124,12 @@ const { densityClass } = useDensity(props);
     font-family: var(--onyx-font-family);
     color: var(--onyx-color-text-icons-neutral-intense);
     text-align: left;
+    contain: paint;
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:127]: will this make trouble in the future e.g. with a dropdown opening from within a cell?

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx [hunk 2]
```diff
@@ -55,4 +55,22 @@ test.describe("Screenshot tests", () => {
       </OnyxTable>
     ),
   });
+
+  executeMatrixScreenshotTest({
+    name: "Table (hover styles)",
+    columns: ["default", "striped"],
+    rows: ["row-hover", "column-hover"],
+    // TODO: remove when contrast issues are fixed in https://github.com/SchwarzIT/onyx/issues/410
+    disabledAccessibilityRules: ["color-contrast"],
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx:64]: fyi in dark mode the hover color is nearly invisible. maybe point that out to Jannick before we go beta?

**Review by BoppLi — APPROVED** [review]

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx (hunk 2 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.ct.tsx:64]: would be fine to add this info e.g. as checkbox to #410 I think :)

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue (hunk 1 above)
Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:127]: I think we can only prevent this if we know the exact table height so we can size the column hover effect without `contain: paint` or `overflow: hidden`...but this means we propably need JavaScript...

Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:127]: @BoppLi I am not sure if I should take care of this in this PR or if we check this later or we might even place the dropdown absolute idk

Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:127]: it's fine for me. let's try to remember it when it comes to it. 👍

**Pull request merged**